packages/calcite-components/src/components/tile/tile.tsx

```diff
@@ -107,29 +240,112 @@ export class Tile implements InteractiveComponent {
     updateHostInteraction(this);
   }
 
+  async componentWillLoad(): Promise<void> {
+    setUpLoadableComponent(this);
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Event Listeners
+  //
+  //--------------------------------------------------------------------------
+
+  @Listen("keydown")
+  keyDownHandler(event: KeyboardEvent): void {
+    if (event.target === this.el) {
+      switch (event.key) {
+        case " ":
+        case "Enter":
+          this.handleSelectEvent();
+          event.preventDefault();
+          break;
+        case "ArrowDown":
+        case "ArrowLeft":
+        case "ArrowRight":
+        case "ArrowUp":
+        case "Home":
+        case "End":
+          this.calciteInternalTileKeyEvent.emit(event);
+          event.preventDefault();
+          break;
+      }
+    }
+  }
+
   // --------------------------------------------------------------------------
   //
   //  Render Methods
   //
   // --------------------------------------------------------------------------
 
-  renderTile(): VNode {
-    const { icon, heading, description, iconFlipRtl } = this;
-    const isLargeVisual = heading && icon && !description;
+  renderSelectionIcon(): VNode {
+    const { selected, selectionAppearance, selectionMode } = this;
+    if (selectionAppearance === "icon" && selectionMode !== "none") {
+      return (
+        <calcite-icon
+          class={CSS.selectionIcon}
+          icon={
+            selected
+              ? selectionMode === "multiple"
+                ? ICONS.selectedMultiple
+                : ICONS.selectedSingle
+              : selectionMode === "multiple"
+                ? ICONS.unselectedMultiple
+                : ICONS.unselectedSingle
+          }
+          scale="s"
+        />
+      );
+    }
+    return;
+  }
 
+  renderTile(): VNode {
+    const { description, disabled, heading, icon, iconFlipRtl, interactive, selectionMode } = this;
+    const isLargeVisual = heading && icon && !Boolean(description);
+    const disableInteraction = disabled || Boolean(this.href) || (!disabled && !interactive);
```

Comment:
You could forgo using `disabled` here since the `interactive` utils will prevent the component from receiving focus.